controlplane/api/v1alpha1/rke2controlplane_types.go

```diff
@@ -17,48 +17,270 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"github.com/rancher-sandbox/cluster-api-provider-rke2/bootstrap/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// Rke2ControlPlaneSpec defines the desired state of Rke2ControlPlane
-type Rke2ControlPlaneSpec struct {
+// RKE2ControlPlaneSpec defines the desired state of RKE2ControlPlane
+type RKE2ControlPlaneSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of Rke2ControlPlane. Edit rke2controlplane_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// ServerConfig specifies configuration for the agent nodes.
```

Comment:
I think we need to have a field of type `bootstrapv1.RKE2ConfigSpec` in this struct

Reply:
Or we inline the struct instead of using a struct field. Either way we should include it.

Reply:
Or we move fields like version, files, PreRKE2Commands to the common struct.

Reply:
Done!